frontend/src/App.js

```diff
@@ -1,15 +1,150 @@
 import React, {useState, useEffect} from 'react';
+import dotenv from "dotenv";
 import './App.css';
+// import GitHubLogin from 'react-github-login';
+
+
+dotenv.config()
+
+
 
 const App = () => {
   const [user, setUser] = useState(null);
+  const [token, setToken] = useState(null);
+
 
   useEffect(() => {
-    fetch('/api/me')
-      .then(response => response.json())
-      .then(setUser);
+
+    //event listener for getting data back from popup
+
+    window.addEventListener('message', function(event){
+
+      // checking access code available or not
+
+        if(event.data != "" || event.data !== undefined){
+      
+          // storing code 
+          let code = event.data;
+          
+          // create data variable to send in api
+
+          let data = {
+            code: code
+          };
+
+          // api for getting jwt token from backend api
+
+          fetch('/api/github/token', {
+                method: 'POST',
+                headers: {
+                  'Accept': 'application/json',
+                  'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(data)
+            }).then(function(response) {
```

Comment:
Please use arrow notation where possible, e.g. `.then(response => response.json())`. Thank you.